test/file.jl

```diff
@@ -215,6 +215,15 @@ close(s)
 # This section tests temporary file and directory creation.           #
 #######################################################################
 
+@testset "quoting filenames" begin
+    try
+        open("foo bar")
+    catch e
+        isa(e, SystemError) || rethrow(e)
+        @test sprint(showerror, e) == "SystemError: opening file \"foo bar\": No such file or directory"
+    end
+end
```

Comment:
maybe adding the following marks to make sure it threw an error (I've left them left-indented for comparison, but you should indent them correctly):
```julia
@testset "quoting filenames" begin
@test     try
         open("foo bar")
false
     catch e
         isa(e, SystemError) || rethrow(e)
         @test sprint(showerror, e) == "SystemError: opening file \"foo bar\": No such file or directory"
true
     end
 end
```

also, I recommend making the file name `this file is not expected to exist` instead of `foo bar`